src/blocks/kdeconnect.rs

```diff
@@ -129,7 +143,31 @@ pub async fn run(config: Config, mut api: CommonApi) -> Result<()> {
                     }
                 }
 
-                let notif_count = device.notifications().await.unwrap_or(0);
+                if let Ok(cellular_network_type) = cellular_network_type {
+                    // network strength is 0..=4 from docs of
+                    // kdeconnect/plugins/connectivity-report, and I
+                    // got -1 for disabled SIM (undocumented)
+                    let cell_network_percent = (cellular_network_strength.clamp(0, 4) * 25) as f64;
+                    values.insert(
+                        "network_icon".into(),
+                        Value::icon(api.get_icon_in_progression(
+                            "net_cellular",
+                            (cellular_network_strength + 1).clamp(0, 5) as f64 / 5.0,
+                        )?),
+                    );
+                    values.insert(
+                        "network_strength".into(),
+                        Value::percents(cell_network_percent),
+                    );
+
+                    if cellular_network_strength <= 0 {
+                        widget.state = State::Critical;
```

Comment:
it would be really nice if things like this could be included in the changelog, or at least in the documentation :) I just spent half an hour trying to figure out why my kdeconnect block was yelling at me for my phone only having 75% battery remaining, because I didn't realise it would now yell at me if my phone is not connected to the cell network. (incidentally, the phone _is_ connected to the cell network, kdeconnect just doesn't know that.)

I also don't see a way of disabling this other than patching i3status-rust – that's fine, I do that anyway, but again it would be nice if that could be mentioned somewhere.

Reply:
So it does that even when you don't use the cellular network plugin from kdeconnect? if that's the case then it'd be a bug. 

IF you don't use kdeconnect network plugin then that code line should be ignored.

Reply:
I'd tried it before, but I tried it again, and if you uncheck the connectivity monitor the block simply disappears.

![image](https://github.com/greshake/i3status-rust/assets/46195892/b4721092-861e-4bc8-8d99-5b88d94d5ca5)
![image](https://github.com/greshake/i3status-rust/assets/46195892/e1e309d9-bd55-4cdd-9de7-da5ff16e9d2f)

![image](https://github.com/greshake/i3status-rust/assets/46195892/ee070219-198b-453e-8a05-ba634fdd08ae)
![image](https://github.com/greshake/i3status-rust/assets/46195892/b001fc8f-698c-4e70-a7ad-4d55f2a0733e)

Reply:
Also an option can be added to select the source of block's state: battery, network or both.

Reply:
ahh, cool, I didn't realise there was a kdeconnect plugin I could disable. thanks :)